config/config.nims

```diff
@@ -14,3 +14,11 @@ when defined(nimStrictMode):
     # switch("hint", "ConvFromXtoItselfNotNeeded")
     switch("hintAsError", "ConvFromXtoItselfNotNeeded")
     # future work: XDeclaredButNotUsed
+
+when (NimMajor, NimMinor) == (1,6) or (NimMajor, NimMinor) <= (1,4):
+  discard
+else:
+  # devel gets bugfixes and some preview flags right away; next release gets bugfixes via a preview flag.
+  # these can be overridden using user/project/cmdline flags using `switch("undef", "nimPreviewX")`
+  # other `nimPreview` switches can go here, as needed.
+  switch("define", "nimPreviewDotLikeOps")
```

Comment:
I think we should instead have a todo for 1.7.x where we go through a list of preview options to consciously sanctionize new features we want to include for 1.7.x. The form of automization that you do here seems not wise.

Reply:
the RFC was marked as accepted so it seems fair to make turn this on (except for 1.6); I can change the surrounding comment but the comment as written doesn't imply that preview flags get turned on automatically, it says:

`some preview flags` and `can go here, as needed.`

which clearly states that these are turned on on a case by case basis.
Waiting for 1.8 to turn this particular feature on serves no benefit, it only makes it harder to use this feature in library code.

Reply:
`-d:enablePreview18` sounds like a reasonable catch-all

Reply:
Please do as I said, I can only repeat my argument. We don't automate feature activations.

Reply:
done for the sake of moving on with this PR, but I still don't know what you mean by `automate feature activations` as I explained above; there's nothing automatic about hand-selected preview flags being turned on.

Reply:
How can we ensure it is enabled after Nim 1.6 is released? What if it is forgotten?

Reply:
Before every release we follow https://github.com/nim-lang/Nim/wiki/Creating-a-release and it's a living document. Feel free to add a note.

Reply:
I would recommend to create release/release.nim

```nim
import std/private/since

when (NimMajor, NimMinor) > (1, 6):
  doAssert defined(nimPreviewDotLikeOps)
```

Reply:
if the criterion is > (1,6) a simpler way is:
```nim
# in config/config.nims:
when (NimMajor, NimMinor) > (1, 6):
  switch("define", "nimPreviewDotLikeOps")
```

Reply:
Looks good to me. Anyway we need a way to catch/automate it.

Reply:
If we need

> more concretely, that means that the feature lives in devel but is disabled. then 1.6 is released with the feature disabled. then devel is switched to enabled the flag. then a decision is made whether to enable the flag in 1.8+.

```nim
when (NimMajor, NimMinor) > (1, 6): # ref https://github.com/nim-lang/Nim/pull/18711
  doAssert defined(nimPreviewDotLikeOps)

when (NimMajor, NimMinor) == (1, 8):
  doAssert not defined(nimPreviewDotLikeOps)
```

Reply:
> How can we ensure it is enabled after Nim 1.6 is released? What if it is forgotten?

this sounds a bit like a non-problem - ie if it's forgotten, the feature is clearly not important or impactful enough that somebody cares/bothers, and one might want to reconsider its utility.

the other thing to consider is that the unit test suite should run _with_ and _without_ the flag, at least a subset of the tests where the old and new versions of the behaviour are touched / stressed: the way to do this is to run the targeted subset with the feature enabled while the feature is disabled by default, then run a targeted subset of tests for the old behaviour once the flag is stripped.

This also brings us back to a critical point: the more care is taken with backwards compatibility, the fewer problems of this nature there are and the easier it is to produce a new release - the complexity of introducing the change is merely being aligned with the effect it will have on existing code.

Reply:
> this sounds a bit like a non-problem - ie if it's forgotten, the feature is clearly not important or impactful enough that somebody cares/bothers, and one might want to reconsider its utility.

It may be true. 

IMO Nim language is a small community and only few people(araq and miran, other maintainers may leave Nim at any time) maintain it. People who like this feature may not know when and whether to change the flag. And people who proposal the feature may leave Nim.
 
Anyway it is nice to make maintenance easier.

Reply:
what would be really interesting is a warning that detects the difference in parsing and warns the user (this would be implementable in devel/next release already)